### PR TITLE
[wip] use copier for npe2 create

### PR DESCRIPTION
**Pull request by tlambert03**
Been playing with using [copier](https://copier.readthedocs.io/en/stable/) as a replacement for cookiecutter.  Generally much happier with it, mostly because of support for 1) conditionals in the questions, 2) much better updating when the upstream template changes 3) validation (after [my PR merges](https://github.com/copier-org/copier/pull/719))

I built a copier template here: https://github.com/tlambert03/napari-plugin-template

this PR would address #54 and add an `npe2 create` command that would bootstrap a new plugin, as @nclack proposed.  Most of the work will be in tailoring that plugin template repo as we want it.  and I also think we should create an actual webpage documentation for it.

if we eventually wanted to go even farther than copier, we could just keep the command

### CODE REVIEW

File: npe2/cli.py [hunk 1]
```diff
@@ -522,5 +522,34 @@ def compile(
...
+@app.command()
+def create(plugin_name: str):
+    """Create a new napari plugin"""
+    import subprocess
+    from shutil import which
+
+    if not which("copier"):
+        text = typer.style(
+            "This command requires the `copier` package.\n"
+            "Install it now into your current environment? [y/N]",
```
Comment (nclack) [inline npe2/cli.py:534]:
```suggestion
            "Install it now into your current environment with pip? [y/N]",
```

**Review by nclack — APPROVED** [review]
Running through the interactive prompts myself, there were usability improvements but there are a few rough edges too.  Those don't need to be addressed here, but I was wondering if you also felt there were rough edges.

Just being able to do `npe2 create my-plugin` feels so so much better to me than having to look up the cookiecutter template. We could even think about having different kinds of templates ('bare minimum' vs 'everything but the kitchen sink')

There's a lot I don't really like about cookiecutter, so overall this feels like a really nice improvement.

**Comment by tlambert03** [comment]
> there were usability improvements but there are a few rough edges too.

I agree, did you write them down? 😅

**Comment by nclack** [comment]
> > there were usability improvements but there are a few rough edges too.
> 
> I agree, did you write them down? 😅

I did :) Off the top of my head:

* For string defaults like the author field, had to delete each char and then enter the non-default test manually.
* The installed paths reported at the end didn't include the name of the source root. For example, it would say "created 'src/napari.yaml'" instead of "created my-silly-plugin/src/napari.yml".

**Comment by tlambert03** [comment]
> had to delete each char and then enter the non-default test manually.

**Comment by nclack** [comment]
> I might leave this PR as WIP for a bit until that template is ready/better

up to you.  I think this is gtg and the template can get iterated on later.

**Pull request closed without merging**